feedback/invalid_token.go

```diff
@@ -145,23 +144,29 @@ func (i *InvalidTokenHandler) processMessages() {
 	flushTicker := time.NewTicker(i.flushTime)
 	defer flushTicker.Stop()
 
-	for i.run {
+	for {
 		select {
 		case tk, ok := <-i.InChan:
 			if ok {
+				i.BufferLock.Lock()
 				i.Buffer = append(i.Buffer, tk)
+				i.BufferLock.Unlock()
 
+				i.BufferLock.Lock()
```

Comment:
Why we don't use the same `.Lock()` + `.Unlock` block for appending and clearing the buffer? Is `i.deleteTokens` a costly operation?

Reply:
No actual reason. Set everything to one lock/unlock.